source/common/secret/secret_manager_impl.cc

```diff
@@ -37,5 +38,35 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::createInlineTlsCertific
   return std::make_shared<TlsCertificateConfigProviderImpl>(tls_certificate);
 }
 
+void SecretManagerImpl::removeDynamicSecretProvider(const std::string& map_key) {
+  ENVOY_LOG(debug, "Unregister secret provider. hash key: {}", map_key);
+
+  ASSERT(dynamic_secret_providers_.erase(map_key) == 1);
```

Comment:
This should be `RELEASE_ASSERT`; otherwise in `opt` builds, this entire line disappears.

Reply:
Thanks for pointing that out. Fixed.